orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/KatoService.java

```diff
@@ -0,0 +1,122 @@
+package com.netflix.spinnaker.orca.clouddriver;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.hash.Hashing;
+import com.netflix.spinnaker.kork.core.RetrySupport;
+import com.netflix.spinnaker.kork.exceptions.IntegrationException;
+import com.netflix.spinnaker.orca.ExecutionContext;
+import com.netflix.spinnaker.orca.clouddriver.model.OperationContext;
+import com.netflix.spinnaker.orca.clouddriver.model.SubmitOperationResult;
+import com.netflix.spinnaker.orca.clouddriver.model.Task;
+import com.netflix.spinnaker.orca.clouddriver.model.TaskId;
+import com.netflix.spinnaker.orca.jackson.OrcaObjectMapper;
+import java.io.InputStream;
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Map;
+import javax.annotation.Nonnull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import retrofit.client.Response;
+
+public class KatoService {
+
+  private final Logger log = LoggerFactory.getLogger(getClass());
+
+  private final KatoRestService katoRestService;
+  private final CloudDriverTaskStatusService cloudDriverTaskStatusService;
+  private final RetrySupport retrySupport;
+  private final ObjectMapper objectMapper;
+
+  public KatoService(
+      KatoRestService katoRestService,
+      CloudDriverTaskStatusService cloudDriverTaskStatusService,
+      RetrySupport retrySupport,
+      ObjectMapper objectMapper) {
+    this.katoRestService = katoRestService;
+    this.cloudDriverTaskStatusService = cloudDriverTaskStatusService;
+    this.retrySupport = retrySupport;
+    this.objectMapper = objectMapper;
+  }
+
+  public TaskId requestOperations(Collection<? extends Map<String, Map>> operations) {
+    return retrySupport.retry(
+        () -> katoRestService.requestOperations(requestId(operations), operations),
+        3,
+        Duration.ofSeconds(1),
+        false);
+  }
+
+  public TaskId requestOperations(
+      String cloudProvider, Collection<? extends Map<String, Map>> operations) {
+    return retrySupport.retry(
+        () -> katoRestService.requestOperations(requestId(operations), cloudProvider, operations),
+        3,
+        Duration.ofSeconds(1),
+        false);
+  }
+
+  public SubmitOperationResult submitOperation(
+      @Nonnull String cloudProvider, OperationContext operation) {
+    Response response =
+        katoRestService.submitOperation(
+            requestId(operation), cloudProvider, operation.getOperationType(), operation);
+
+    TaskId taskId;
+    try {
+      InputStream body = null;
+      try {
+        body = response.getBody().in();
+      } finally {
+        if (body != null) {
+          body.close();
```

Comment:
This is how it was done before but it seems weird and wrong, the InputStream is being closed before it is read by the objectMapper. Also it seems like the whole thing should be switched to use a try with resources.

Reply:
Yeah, this method is honestly only being used from a stage/task that is completely unused (`CloudOperationStage`) so I didn't spend anytime thinking about this (pretty sure eventually we would just delete `CloudOperationStage`).

May as well clean it up though I suppose.